tests/fixture/tmpnet/kube.go

```diff
@@ -0,0 +1,299 @@
+func WaitForNodeHealthy(
+	ctx context.Context,
+	// TODO(marun) Use an actual logger
+	logFunc func(format string, args ...interface{}),
+	kubeconfig *restclient.Config,
+	namespace string,
+	podName string,
+	healthCheckInterval time.Duration,
+	out io.Writer,
+	outErr io.Writer,
+) (ids.NodeID, error) {
+	// A forwarded connection enables connectivity without exposing the node external to the kube cluster
+	localPort, localPortStopChan, err := enableLocalForwardForPod(
+		kubeconfig,
+		namespace,
+		podName,
+		config.DefaultHTTPPort,
+		out,
+		outErr,
+	)
+	if err != nil {
+		return ids.NodeID{}, fmt.Errorf("failed to enable local forward for pod: %w", err)
+	}
+	defer close(localPortStopChan)
+	localNodeURI := fmt.Sprintf("http://127.0.0.1:%d", localPort)
+
+	// TODO(marun) A node started with tmpnet should know the node ID before start
+	infoClient := info.NewClient(localNodeURI)
+	bootstrapNodeID, _, err := infoClient.GetNodeID(ctx)
+	if err != nil {
+		return ids.NodeID{}, fmt.Errorf("failed to retrieve node bootstrap ID: %w", err)
+	}
+	if err := wait.PollImmediateInfinite(healthCheckInterval, func() (bool, error) {
+		healthReply, err := CheckNodeHealth(ctx, localNodeURI)
+		if errors.Is(ErrUnrecoverableNodeHealthCheck, err) {
+			return false, err
+		} else if err != nil {
+			// Error is potentially recoverable - log and continue
+			logFunc("Error checking node health: %v", err)
+			return false, nil
+		}
+		return healthReply.Healthy, nil
+	}); err != nil {
+		return ids.NodeID{}, fmt.Errorf("failed to wait for node to report healthy: %w", err)
+	}
+
+	return bootstrapNodeID, nil
+}
+
+// WaitForPodCondition watches the specified pod until the status includes the specified condition.
+func WaitForPodCondition(ctx context.Context, clientset *kubernetes.Clientset, namespace string, podName string, conditionType corev1.PodConditionType) error {
+	return WaitForPodStatus(
+		ctx,
+		clientset,
+		namespace,
+		podName,
+		func(status *corev1.PodStatus) bool {
+			for _, condition := range status.Conditions {
+				if condition.Type == conditionType && condition.Status == corev1.ConditionTrue {
+					return true
+				}
+			}
+			return false
+		},
+	)
+}
+
+// WaitForPodStatus watches the specified pod until the status is deemed acceptable by the provided test function.
+func WaitForPodStatus(
+	ctx context.Context,
+	clientset *kubernetes.Clientset,
+	namespace string,
+	name string,
+	acceptable func(*corev1.PodStatus) bool,
+) error {
+	watch, err := clientset.CoreV1().Pods(namespace).Watch(ctx, metav1.SingleObject(metav1.ObjectMeta{Name: name}))
+	if err != nil {
+		return fmt.Errorf("failed to initiate watch of pod %s/%s: %w", namespace, name, err)
+	}
+
+	for {
+		select {
+		case event := <-watch.ResultChan():
+			pod, ok := event.Object.(*corev1.Pod)
+			if !ok {
+				continue
+			}
+
+			if acceptable(&pod.Status) {
+				return nil
+			}
+		case <-ctx.Done():
+			return errors.New("timeout waiting for pod readiness")
```

Comment:
We used to return `ctx.Err()` here. Why aren't we anymore?
```suggestion
			return fmt.Errorf("timeout waiting for pod readiness: %w", ctx.Err())
```